nevergrad/parametrization/core.py

```diff
@@ -0,0 +1,497 @@
+        if self._subparameters is not None and self.subparameters:
+            substr = f"[{self.subparameters._get_parameters_str()}]"
+        return f"{self._get_name()}" + substr
+
+    @name.setter
+    def name(self, name: str) -> None:
+        self.set_name(name)  # with_name allows chaining
+
+    def __repr__(self) -> str:
+        return f"{self.name}:{self.value}".replace(" ", "").replace("\n", "")
+
+    def set_name(self: P, name: str) -> P:
+        """Sets a name and return the current instrumentation (for chaining)
+
+        Parameter
+        ---------
+        name: str
+            new name to use to represent the Parameter
+        """
+        self._name = name
+        return self
+
+    # %% Constraint management
+
+    def satisfies_constraint(self) -> bool:
+        """Whether the instance satisfies the constraints added through
+        the "register_cheap_constraint" method
+
+        Returns
+        -------
+        bool
+            True iff the constraint is satisfied
+        """
+        if not self._constraint_checkers:
+            return True
+        val = self.value
+        return all(func(val) for func in self._constraint_checkers)
+
+    def register_cheap_constraint(self, func: t.Callable[[t.Any], bool]) -> None:
+        """Registers a new constraint on the parameter values.
+
+        Parameter
+        ---------
+        func: Callable
+            function which, given the value of the instance, returns whether it satisfies the constraint.
+
+        Note
+        - this is only for checking after mutation/recombination/etc if the value still satisfy the constraints.
+          The constraint is not used in those processes.
+        - constraints should be fast to compute.
+        """
+        if getattr(func, "__name__", "not lambda") == "<lambda>":  # LambdaType does not work :(
+            warnings.warn("Lambda as constraint is not advice because it may not be picklable")
+        self._constraint_checkers.append(func)
+
+    # %% random state
+
+    @property
+    def random_state(self) -> np.random.RandomState:
+        """Random state the instrumentation and the optimizers pull from.
+        It can be seeded/replaced.
+        """
+        if self._random_state is None:
+            # use the setter, to make sure the random state is propagated to the variables
+            seed = np.random.randint(2 ** 32, dtype=np.uint32)
+            self._set_random_state(np.random.RandomState(seed))
+        assert self._random_state is not None
+        return self._random_state
+
+    @random_state.setter
+    def random_state(self, random_state: np.random.RandomState) -> None:
+        self._set_random_state(random_state)
+
+    def _set_random_state(self, random_state: np.random.RandomState) -> None:
+        self._random_state = random_state
+        if self._subparameters is not None:
+            self.subparameters._set_random_state(random_state)
+
+    def spawn_child(self: P) -> P:
```

Comment:
Shouldn't this be in `BaseParameter`?

Reply:
Well... "BaseParameter" does not really exit actually, it's just Parameter split in two for readability... I need to make it more explicit in the comments I think

Reply:
If the whole world is against me, then so be it lol, it's now merged